src/lib.rs

```diff
@@ -40,26 +40,26 @@ pub use serde::{de::DeserializeOwned, Serialize};
 /// The precedence in the case where multiple errors are applicable
 /// simultaneously is from high to low:
 ///
-/// `Internal > Absent > TooLong > NotFound > Inner > PostDeserialization`
+/// `Absent > TooShort > NotFound > TooLong > Inner > PostDeserialization`
 /// before any `Ok`.
 #[non_exhaustive]
 #[derive(Debug, Copy, Clone, PartialEq, Eq)]
 pub enum Error<E> {
-    /// A name was not found or an index was too large or invalid.
-    NotFound(usize),
-
-    /// The names/indices are valid, but do not exist at runtime.
+    /// The key is valid, but does not exist at runtime.
     ///
     /// This is the case if a deferred [core::option::Option] or [Option]
     /// is `None` at runtime.
     Absent(usize),
 
-    /// The names/indices are valid, but there is trailing data at the end.
-    TooLong(usize),
-
-    /// The names/indices are valid, but end early and do not reach a leaf.
+    /// The key ends early and does not reach a leaf node.
     TooShort(usize),
 
+    /// A key was not found, is too large or invalid.
```

Comment:
I'm not sure what "Too large" is supposed to mean? Also, what's the difference between "invalid" and "not found"?